api/addContact.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+header("Content-Type: application/json; charset=UTF-8");
+header("Access-Control-Allow-Methods: POST");
+
+require_once 'vendor/autoload.php';
+$dotenv = Dotenv\Dotenv::createImmutable(__DIR__);
+$dotenv->load();
+
+function getConnection()
+{
+    $host = $_ENV['DB_HOST'];  // Fetch from .env
+    $db_name = $_ENV['DB_NAME']; // Fetch from .env
+    $username = $_ENV['DB_USER']; // Fetch from .env
+    $password = $_ENV['DB_PASS']; // Fetch from .env
+
+    try {
+        $conn = new PDO("mysql:host=$host;dbname=$db_name;charset=utf8mb4", $username, $password);
+        $conn->setAttribute(PDO::ATTR_ERRMODE, PDO::ERRMODE_EXCEPTION);
+        return $conn;
+    } catch (PDOException $exception) {
+        http_response_code(500);
+        echo json_encode(["message" => "Database connection error: " . $exception->getMessage()]);
+        die();
+    }
+}
+
+function checkIfContactExists($uid, $phoneNumber, $email)
+{
+    $conn = getConnection();
+
+    // check both phone number and email for a given uid
+    $query = "SELECT * FROM Contact WHERE uid = :uid AND (phoneNumber = :phoneNumber OR email = :email) LIMIT 1";
+    $stmt = $conn->prepare($query);
+
+    $stmt->bindParam(':uid', $uid);
+    $stmt->bindParam(':phoneNumber', $phoneNumber);
+    $stmt->bindParam(':email', $email);
+
+    $stmt->execute();
+    return $stmt->fetch(PDO::FETCH_ASSOC);
+}
+
+$input = json_decode(file_get_contents("php://input"), true);
+
+if (
```

Comment:
Are we strictly enforcing a phone number in order to create a new contact? If not we don't need to check for a phone number upon adding a contact.

Reply:
I think we are enforcing everything is order to create a contact.